### ISSUE TITLE
[BUG] another small traits dict -> str type error in cbeditors

### ISSUE THREAD

**Issue by barentine** [python-microscopy/python-microscopy#404]
**Describe the bug**
```
Traceback (most recent call last):
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\matplotlib\cbook\__init__.py", line 196, in process
    func(*args, **kwargs)
  File "c:\users\bergamot\code\python-microscopy\PYME\recipes\recipeGui.py", line 533, in OnPick
    outp = self.recipes.activeRecipe.namespace[k]
KeyError: 'overlap_identified'
Traceback (most recent call last):
  File "c:\users\bergamot\code\python-microscopy\PYME\ui\custom_traits_editors.py", line 48, in update_editor
    n = choices.index(self.value)
ValueError: {'filtered': 'roi_locations', 'overlap_identified': 'og_detected'} is not in list

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\matplotlib\cbook\__init__.py", line 196, in process
    func(*args, **kwargs)
  File "c:\users\bergamot\code\python-microscopy\PYME\recipes\recipeGui.py", line 531, in OnPick
    self.configureModule(k)
  File "c:\users\bergamot\code\python-microscopy\PYME\recipes\recipeGui.py", line 553, in configureModule
    k.edit_no_invalidate(handler=MControl())
  File "c:\users\bergamot\code\python-microscopy\PYME\recipes\base.py", line 260, in edit_no_invalidate
    self.edit_traits(*args, kind='modal', **kwargs)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traits\has_traits.py", line 1678, in edit_traits
    args,
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\view.py", line 462, in ui
    ui.ui(parent, kind)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\ui.py", line 246, in ui
    self.rebuild(self, parent)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\toolkit.py", line 133, in ui_modal
    ui_modal.ui_modal(ui, parent)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_modal.py", line 44, in ui_modal
    _ui_dialog(ui, parent, BaseDialog.MODAL)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_modal.py", line 63, in _ui_dialog
    BaseDialog.display_ui(ui, parent, style)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_base.py", line 71, in display_ui
    ui.owner.init(ui, parent, style)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_modal.py", line 137, in init
    sw = panel(ui, window)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_panel.py", line 272, in panel
    panel, content[0], ui
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_panel.py", line 431, in fill_panel_for_group
    panel, group, ui, suppress_label, is_dock_window, create_panel
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_panel.py", line 587, in __init__
    self.add_items(content, panel, self.sizer)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_panel.py", line 909, in add_items
    editor.prepare(item_panel)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\editor.py", line 242, in prepare
    self.update_editor()
  File "c:\users\bergamot\code\python-microscopy\PYME\ui\custom_traits_editors.py", line 51, in update_editor
    self.control.SetValue(self.value)
TypeError: ComboBox.SetValue(): argument 1 has unexpected type 'TraitDictObject'
c:\users\bergamot\code\python-microscopy\PYME\IO\DataSources\ArrayDataSource.py:45: FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be interpreted as an array index, `arr[np.array(seq)]`, which will result either in an error or a different result.
  r = self.data.__getitem__(keys)
c:\users\bergamot\code\python-microscopy\PYME\ui\histLimits.py:289: wxPyDeprecationWarning: Call to deprecated item EmptyBitmap. Use :class:`wx.Bitmap` instead
  MemBitmap = wx.EmptyBitmap(s.GetWidth(), s.GetHeight())
Traceback (most recent call last):
  File "c:\users\bergamot\code\python-microscopy\PYME\ui\custom_traits_editors.py", line 48, in update_editor
    n = choices.index(self.value)
ValueError: {'filtered': 'roi_locations', 'overlap_identified': 'og_detected'} is not in list

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\matplotlib\cbook\__init__.py", line 196, in process
    func(*args, **kwargs)
  File "c:\users\bergamot\code\python-microscopy\PYME\recipes\recipeGui.py", line 531, in OnPick
    self.configureModule(k)
  File "c:\users\bergamot\code\python-microscopy\PYME\recipes\recipeGui.py", line 553, in configureModule
    k.edit_no_invalidate(handler=MControl())
  File "c:\users\bergamot\code\python-microscopy\PYME\recipes\base.py", line 260, in edit_no_invalidate
    self.edit_traits(*args, kind='modal', **kwargs)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traits\has_traits.py", line 1678, in edit_traits
    args,
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\view.py", line 462, in ui
    ui.ui(parent, kind)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\ui.py", line 246, in ui
    self.rebuild(self, parent)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\toolkit.py", line 133, in ui_modal
    ui_modal.ui_modal(ui, parent)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_modal.py", line 44, in ui_modal
    _ui_dialog(ui, parent, BaseDialog.MODAL)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_modal.py", line 63, in _ui_dialog
    BaseDialog.display_ui(ui, parent, style)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_base.py", line 71, in display_ui
    ui.owner.init(ui, parent, style)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_modal.py", line 137, in init
    sw = panel(ui, window)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_panel.py", line 272, in panel
    panel, content[0], ui
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_panel.py", line 431, in fill_panel_for_group
    panel, group, ui, suppress_label, is_dock_window, create_panel
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_panel.py", line 587, in __init__
    self.add_items(content, panel, self.sizer)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\wx\ui_panel.py", line 909, in add_items
    editor.prepare(item_panel)
  File "C:\Users\Bergamot\Anaconda2\envs\pyme\lib\site-packages\traitsui\editor.py", line 242, in prepare
    self.update_editor()
  File "c:\users\bergamot\code\python-microscopy\PYME\ui\custom_traits_editors.py", line 51, in update_editor
    self.control.SetValue(self.value)
TypeError: ComboBox.SetValue(): argument 1 has unexpected type 'TraitDictObject'
```

**To Reproduce**
try and open an HDFOutput module in the recipe gui to edit it after it's already set

same error / fix as #351 , will PR when I'm home

**Comment by barentine** [comment]
Should mention the simple fix is just a str typing, but as discussed in #278 the alternatives are to
a) re-jig HDFOutput so that it doesn't need a dictionary input (I suspect this will break some existing recipes?)
or
b) write a custom editor and view specifically for HDFOutput.

How do we feel about these vs dropping in a str typing to preserve the current py2 behavior on py3 and come back to it when we feel like it?

**Comment by barentine** [comment]
temporarily fixed in #405, see #278 for more info
